storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -123,12 +123,13 @@
             port (:port worker)
             storm-cluster-state (:storm-cluster-state worker)
             prev-backpressure-flag @(:backpressure worker)]
-        (if executors 
-          (if (reduce #(or %1 %2) (map #(.get-backpressure-flag %1) executors))
-            (reset! (:backpressure worker) true)   ;; at least one executor has set backpressure
-            (reset! (:backpressure worker) false))) ;; no executor has backpressure set
+        (when executors
+          (reset! (:backpressure worker)
+                  (or @(:transfer-backpressure worker)
```

Comment:
This was actually a bug, and I am going to file a JIRA for this.  Having it reuse the worker backpressure flag, made it turn off backpressure too soon in some cases, and actually have it get stuck off when it should be on in other cases.